### PR TITLE
- adds github action for releasing frontend

### LINKED ISSUE #85: Deploy fronted to github.io-pages
Use some github action / workflow to deploy the app to our `project-leek.github.io` website.

### PR DESCRIPTION
**Pull request by JanEberlein**
- adds build-routine for frontend
closes #85

### CODE REVIEW

File: .github/workflows/release-app.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - run: yarn lint
+
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
```
Comment (anbraten) [inline .github/workflows/release-app.yml:35]: The checkout, caching, node setup has to be done in each job. 

```suggestion

      - name: Caching
        uses: actions/cache@v2
        with:
          path: ${{ env.YARN_CACHE_FOLDER }}
          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            ${{ runner.os }}-yarn-

      - name: Set up Node
        uses: actions/setup-node@v1
        with:
          node-version: '12.x'

```

**Review by anbraten — CHANGES_REQUESTED** [review]

File: .github/workflows/release-app.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          path: ${{ env.YARN_CACHE_FOLDER }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Set up Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - run: yarn --frozen-lockfile
+      - run: yarn lint
```
[the remark this answers is not in the document]
Comment (JanEberlein) [reply .github/workflows/release-app.yml:26]: Good idea 👍

File: .github/workflows/release-app.yml (hunk 1 above)
Comment (JanEberlein) [reply .github/workflows/release-app.yml:35]: ok

File: .github/workflows/release-app.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Install and Build 🔧
+        run: |
+          yarn --frozen-lockfile
+          yarn build
```
[the remark this answers is not in the document]
Comment (julianhahn) [reply .github/workflows/release-app.yml:39]: @JanEberlein I'll review after your changes are applied

**Review by anbraten — APPROVED** [review]

**Review by julianhahn — APPROVED** [review]
looking good

**Pull request merged**